boards/arm/mimxrt1060_evk/pinmux.c

```diff
@@ -58,6 +58,26 @@ static int mimxrt1060_evk_init(struct device *dev)
 			    IOMUXC_SW_PAD_CTL_PAD_DSE(6));
 #endif
 
+#ifdef CONFIG_I2C_1
+	/* LPI2C1 SCL, SDA */
+	IOMUXC_SetPinMux(IOMUXC_GPIO_AD_B1_00_LPI2C1_SCL, 1);
+	IOMUXC_SetPinMux(IOMUXC_GPIO_AD_B1_01_LPI2C1_SDA, 1);
+
+	IOMUXC_SetPinConfig(IOMUXC_GPIO_AD_B1_00_LPI2C1_SCL,
+			    IOMUXC_SW_PAD_CTL_PAD_PUS(3) |
+			    IOMUXC_SW_PAD_CTL_PAD_PKE_MASK |
+			    IOMUXC_SW_PAD_CTL_PAD_ODE_MASK |
+			    IOMUXC_SW_PAD_CTL_PAD_SPEED(2) |
+			    IOMUXC_SW_PAD_CTL_PAD_DSE(6));
```

Comment:
@MaureenHelm I haven't extensively used this MCU yet at a low level, but this is **disabling** the internal pull up, right, since `0` to `PUE` + `1` to `PKE` = keeper mode, so no internal pull up?

Looking at the reference manual this seems to be:

- `IOMUXC_SW_PAD_CTL_PAD_PUS(3)` = 22K Ohm pull up
- `IOMUXC_SW_PAD_CTL_PAD_PKE_MASK` = Pull/keep enabled
- `IOMUXC_SW_PAD_CTL_PAD_ODE_MASK` = Open drain enabled
- `IOMUXC_SW_PAD_CTL_PAD_SPEED(2)` = Medium (100MHz)
- `IOMUXC_SW_PAD_CTL_PAD_DSE(6)` = Drive strength div by 6

Would it ever make sense to enable the internal 22K pullup for SCL and SDA as a config option (I think setting `PUE`to `1`)?

22K is likely too high for I2C, but just wondering if there is a valid use case where the board doesn't have pullups populated and you need to enable the internal 22K ones to get the I2C bus working? Probably not, and you generally want something in the 4.7K range anyway with I2C.

Reply:
> 22K is likely too high for I2C, but just wondering if there is a valid use case where the board doesn't have pullups populated and you need to enable the internal 22K ones to get the I2C bus working? Probably not, and you generally want something in the 4.7K range anyway with I2C.

No, I don't think it makes sense since this pinmux config code is part of the board-level init. I suppose maybe you might use the internal pullup on a custom board if that board didn't have external pullups, but that would be configured in a different pinmux.c.